beacon-chain/rpc/eth/rewards/handlers.go

```diff
@@ -166,22 +178,318 @@ func (s *Server) BlockRewards(w http.ResponseWriter, r *http.Request) {
+	if err != nil {
+		errJson := &network.DefaultErrorJson{
+			Message: "Could not get state for epoch's starting slot: " + err.Error(),
+			Code:    http.StatusInternalServerError,
+		}
+		network.WriteError(w, errJson)
+		return
+	}
+
+	allVals, bal, err := altair.InitializePrecomputeValidators(r.Context(), st)
+	if err != nil {
+		errJson := &network.DefaultErrorJson{
+			Message: "Could not initialize precompute validators: " + err.Error(),
+			Code:    http.StatusBadRequest,
+		}
+		network.WriteError(w, errJson)
+		return
+	}
+	allVals, bal, err = altair.ProcessEpochParticipation(r.Context(), st, bal, allVals)
+	if err != nil {
+		errJson := &network.DefaultErrorJson{
+			Message: "Could not process epoch participation: " + err.Error(),
+			Code:    http.StatusBadRequest,
+		}
+		network.WriteError(w, errJson)
+		return
+	}
+
+	var rawValIds []string
+	if r.Body != http.NoBody {
+		if err = json.NewDecoder(r.Body).Decode(&rawValIds); err != nil {
+			errJson := &network.DefaultErrorJson{
+				Message: "Could not decode validators: " + err.Error(),
+				Code:    http.StatusBadRequest,
+			}
+			network.WriteError(w, errJson)
+			return
+		}
+	}
+	valIndices := make([]primitives.ValidatorIndex, len(rawValIds))
+	for i, v := range rawValIds {
+		index, err := strconv.ParseUint(v, 10, 64)
+		if err != nil {
+			pubkey, err := bytesutil.FromHexString(v)
+			if err != nil || len(pubkey) != fieldparams.BLSPubkeyLength {
+				errJson := &network.DefaultErrorJson{
+					Message: fmt.Sprintf("%s is not a validator index or pubkey", v),
+					Code:    http.StatusBadRequest,
+				}
+				network.WriteError(w, errJson)
+				return
+			}
+			var ok bool
+			valIndices[i], ok = st.ValidatorIndexByPubkey(bytesutil.ToBytes48(pubkey))
+			if !ok {
+				errJson := &network.DefaultErrorJson{
+					Message: fmt.Sprintf("No validator index found for pubkey %#x", pubkey),
+					Code:    http.StatusBadRequest,
+				}
+				network.WriteError(w, errJson)
+				return
+			}
+		} else {
+			if i >= st.NumValidators() {
+				errJson := &network.DefaultErrorJson{
+					Message: fmt.Sprintf("Validator index %d is too large. Maximum allowed index is %d", i, st.NumValidators()-1),
+					Code:    http.StatusBadRequest,
+				}
+				network.WriteError(w, errJson)
+				return
+			}
+			valIndices[i] = primitives.ValidatorIndex(index)
+		}
+	}
+	if len(valIndices) == 0 {
+		valIndices = make([]primitives.ValidatorIndex, len(allVals))
+		for i := 0; i < len(allVals); i++ {
+			valIndices[i] = primitives.ValidatorIndex(i)
+		}
+	}
+	var filteredVals []*precompute.Validator
+	if len(valIndices) == len(allVals) {
+		filteredVals = allVals
+	} else {
+		filteredVals = make([]*precompute.Validator, len(valIndices))
+		for i, valIx := range valIndices {
+			filteredVals[i] = allVals[valIx]
+		}
+	}
+
+	idealRewards := make([]AttReward, 32)
+	idealVals := make([]*precompute.Validator, 32)
+	increment := params.BeaconConfig().EffectiveBalanceIncrement
+	for i := 1; i < 33; i++ {
```

Comment:
Better to use `MaxEffectiveBalance` here and in the previous line to obtain 32 and 32+1

Reply:
`MaxEffectiveBalance` is equal `32 * 1e9`

Reply:
I still don't like these unnamed constants